### PR TITLE
Streaming support

### PR DESCRIPTION
**Pull request by nqn**

### CODE REVIEW

File: log10/load.py [hunk 1]
```diff
@@ -267,6 +268,68 @@ def log_sync(completion_url, func, **kwargs):
...
+            # Log the final result
+            # Create fake response for openai format.
+            response = {
+                "id": self.gpt_id,
+                "object": "completion",
+                "model": self.model,
+                "choices": [
+                    {
+                        "index": 0,
+                        "finish_reason": "stop",
```
Comment (wenzhe-log10) [inline log10/load.py:308]: could max_token reached be the other reason?

File: log10/load.py [hunk 2]
```diff
@@ -267,6 +268,68 @@ def log_sync(completion_url, func, **kwargs):
...
+                if res.status_code != 200:
+                    logger.error(f"LOG10: failed to insert in log10: {self.partial_log_row} with error {res.text}")
+            except Exception as e:
+                traceback.print_tb(e.__traceback__)
+                logging.warn(f"LOG10: failed to log: {e}. Skipping")
+
+            raise se
+
+    def get_final_result(self):
+        return self.final_result
```
Comment (wenzhe-log10) [inline log10/load.py:330]: curious is this been called somewhere now or just a util function? Do we need to check StopIterator or has been raised or some state telling that this is the final results?

**Review by wenzhe-log10 — COMMENTED** [review]
Nice work! Left some comments.
Also could you please add an example either in a doctest (in load.py) or an example in `examples/logging`?

File: log10/load.py [hunk 3]
```diff
@@ -267,6 +268,68 @@ def log_sync(completion_url, func, **kwargs):
...
+
+            try:
+                res = post_request(self.completion_url + "/" + self.completionID, self.partial_log_row)
+                if res.status_code != 200:
+                    logger.error(f"LOG10: failed to insert in log10: {self.partial_log_row} with error {res.text}")
+            except Exception as e:
+                traceback.print_tb(e.__traceback__)
+                logging.warn(f"LOG10: failed to log: {e}. Skipping")
+
+            raise se
```
[the remark this answers is not in the document]
Comment (nqn) [reply log10/load.py:327]: Yep that's right

File: log10/load.py (hunk 2 above)
Comment (nqn) [reply log10/load.py:330]: We can remove it

File: log10/load.py (hunk 1 above)
Comment (nqn) [reply log10/load.py:308]: Let's see if we can get the stop reason from the last chunk above. Will give that a go

Comment (nqn) [reply log10/load.py]: Worked - should have been updated

File: log10/load.py [hunk 4]
```diff
@@ -267,6 +268,68 @@ def log_sync(completion_url, func, **kwargs):
...
+            self.partial_log_row["response"] = json.dumps(response)
+            self.partial_log_row["duration"] = int(time.perf_counter() - self.start_time * 1000)
+
+            try:
+                res = post_request(self.completion_url + "/" + self.completionID, self.partial_log_row)
+                if res.status_code != 200:
+                    logger.error(f"LOG10: failed to insert in log10: {self.partial_log_row} with error {res.text}")
+            except Exception as e:
+                traceback.print_tb(e.__traceback__)
+                logging.warn(f"LOG10: failed to log: {e}. Skipping")
```
[the remark this answers is not in the document]
Comment (wenzhe-log10) [reply log10/load.py:328]: yea, seems there's a mix of logger.error and logger.warn in the load.py. could do a clean up in a follow up PR.

**Review by wenzhe-log10 — APPROVED** [review]
LGTM. please add the example before merging.

**Pull request merged**